assets/js/blocks/featured-product/block.js

```diff
@@ -307,7 +307,7 @@ class FeaturedProduct extends Component {
 										<div
 											className="wc-block-featured-product__description"
 											dangerouslySetInnerHTML={ {
-												__html: product.short_description,
+												__html: 'variation' === product.type ? product.description : product.short_description,
```

Comment:
Variations only have a single description which is a small text area. Whilst it's called 'description' rather than 'short description' in the API, I don't think this would be used for something super long. It's all that variations have available.